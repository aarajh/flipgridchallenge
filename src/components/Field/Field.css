.DefaultFieldClass{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    min-width: 100%;
    padding: 0%;
}

.DefaultInnerClass{
    width: 100%;
    line-height: 1.5em;
    font-size: inherit;
    border-radius: 5px;
    border: 1.5px solid rgba(0,0,0,0.1);
    padding: 5px;
}

.DefaultLabelClass{
    font-weight: 600;
    padding: 5px 0px 5px 0px;
}