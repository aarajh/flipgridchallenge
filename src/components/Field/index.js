import React from 'react';
import './Field.css'
import "typeface-roboto";

const ShowError = error =>(
    <div style={error?{}:{visibility:'hidden'}} className="ErrorField">
        {error}
    </div>
)

const Field = props => (
    <div className={props.className} data-testid={props["main-testid"]}>
        <div className={props.labelClass}>
        {props.label}
        {ShowError(props.error)}
        </div>
    <input 
        data-testid={props["data-testid"]}
        className={props.innerClass}
        style={props.error?{borderColor:"#df0d6f52"}:null}
        type={props.type} 
        name={props.name}
        onChange={props.onChange}
        value={props.value}
        > 
    
    </input>
    </div>
)

Field.defaultProps = {
    label: 'Field',
    name: 'Field',
    type: 'text',
    value: '',
    "main-testid":"field",
    "data-testid": "inputfield",
    defaultValue: '',
    error: null,
    innerClass: 'DefaultInnerClass',
    labelClass: 'DefaultLabelClass',
    className: 'DefaultFieldClass',
    onChange: null,
}

export default Field;