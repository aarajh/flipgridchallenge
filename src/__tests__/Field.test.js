import React from 'react';
import { render,cleanup, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import Field from 'components/Field/';
import renderer from "react-test-renderer"

afterEach(cleanup)

test('renders default Field', () => {
  const { getByTestId } = render(<Field onChange={jest.fn()} />);
  expect(getByTestId("field")).toHaveTextContent("Field");
});

test('renders custom label', () => {
    const { getByTestId } = render(<Field onChange={jest.fn()} label="Name" />);
    expect(getByTestId("field")).toHaveTextContent("Name");
});

//Test mocks updating the Field to check if the onChange function is called.
test('onChange passed works',()=>{
  const onChange = jest.fn();
  const {getByTestId}=render(<Field onChange={onChange}/>);
  fireEvent.change(getByTestId("inputfield"), { target: { value: 'Test Input' }});
  expect(onChange).toHaveBeenCalled();
});
  
test("matches snapshot", ()=>{
    const tree=renderer.create(<Field/>).toJSON();
    expect(tree).toMatchSnapshot()
})