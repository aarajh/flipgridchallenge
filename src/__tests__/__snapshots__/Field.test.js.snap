// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="DefaultFieldClass"
  data-testid="field"
>
  <div
    className="DefaultLabelClass"
  >
    Field
    <div
      className="ErrorField"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    />
  </div>
  <input
    className="DefaultInnerClass"
    data-testid="inputfield"
    name="Field"
    onChange={null}
    style={null}
    type="text"
    value=""
  />
</div>
`;
